components/scream/src/share/field/field.hpp

```diff
@@ -464,6 +468,62 @@ sync_to_dev () const {
   Kokkos::deep_copy(m_view_d,*m_view_h);
 }
 
+template<typename RealType>
+template<HostOrDevice HD>
+void Field<RealType>::
+set_value (const field_type& field_src) {
+  const auto& layout     = get_header().get_identifier().get_layout();
+  const auto& layout_src = field_src.get_header().get_identifier().get_layout();
+  EKAT_REQUIRE_MSG(layout==layout_src,
+       "ERROR: Unable to copy field " + field_src.get_header().get_identifier().name() + 
+          " to field " + get_header().get_identifier().name() + " during initialization.  Layouts don't match.");
```

Comment:
Let's not say "during initialization". We don't know if/when other people might be using this method.